lib/ObjWriter/objwriter.cpp

```diff
@@ -387,14 +389,19 @@ GetSymbolRefExpr(ObjectWriter *OW, const char *SymbolName,
 }
 
 extern "C" void EmitSymbolRef(ObjectWriter *OW, const char *SymbolName,
-                              int Size, bool IsPCRelative, int Delta = 0) {
+                              int Size, bool IsPCRelative, int Delta,
+                              int Kind) {
```

Comment:
The only question I've got is around whether it's a good idea to use the opaque kind and cast to the VariantKind.  It's not quite leaking LLVM types out of the ObjWriter but it's close since you need to keep sources in sync.  Is it worth adding separate entry points for the different kinds with appropriate names?   That has it's own downside in possible number of entrypoints but it would mean that LLVM changes to the relocs only need to fixed on one side.